[CQT-258] Create Merger and SingleQubitGatesMerger classes (#387)
QuTech-Delft/OpenSquirrel: docs/tutorial.md

### Merging single qubit gates

> All single-qubit gates appearing in a circuit can be merged by applying `merge_single_qubit_gates(merger)` to the circuit.

Reviewer: ```suggestion
All single-qubit gates appearing in a circuit can be merged by applying `merge(merger=SingleQubitGatesMerger)` to the circuit.
```

> All single-qubit gates appearing in a circuit can be merged by applying `merge(merger=SingleQubitGatesMerger())` to the circuit.

Reviewer 2: ```suggestion
All single-qubit gates appearing in a circuit can be merged by applying `merge` (with a given instantiation from `Merger`) to the circuit.
```

Reviewer: In the future we may add a two-qubit gate merger as well. So this should be a specific instantiation of the `SingleQubitGatesMerger`, not just any kind of `Merger`

    builder.Rx(0, math.pi / 4)
qc = builder.to_circuit()

> qc.merge_single_qubit_gates(SingleQubitGatesMerger)

Reviewer: ```suggestion
qc.merge(merger=SingleQubitGatesMerger())
```
